backend/src/controllers/sessionsControllers.ts

```diff
@@ -0,0 +1,49 @@
+
+import { Request, Response } from 'express';
+import { createSessionService } from '../services/sessionsServices';
+import jwt from 'jsonwebtoken';
+
+
+export const createSessionController = async(req: Request, res: Response) => {
+
+    try {
+
+        const result = await createSessionService()
```

Comment:
Reading this line I would think that you are instaciating a new instance of `SessionService`.
I would suggest something like `createSession()` or `sessionService.createSession()`. You'd have to adjust your service accordingly.

Reply:
This is an area I find I struggle with in functional programming. The sessionService.createSession() approach makes sense from an OOP perspective.

If I want to stay functional, an issue I find I run into is name conflicts.

For example, I may have a need to create a "createSession" module in the data layer, service layer and also the controller. If I import those within each other, I run into an issue and will need to use an alias, which I find gets very messy and hard to maintain.

The way I've come to avoid alias' is adding a bit more to the name. Eg. createSessionDb() or createSessionService() etc.

I do see your point here, and I agree with you. I'm just not certain a good approach here.

Do you have any resources or keywords I can consider when doing more research in this area? It's been a pain points for me for a long time.

Reply:
You can refactor [sessionsServices.ts](https://github.com/grazirs/movie-matcher/pull/1/files#diff-6c59950cafe2e2af1d5626560e99ec63685ac1118faaf0b1cd61cfc5fe18b4ca) like this:

```
const createSession = async () => {
    // ...
};

export default { createSession };
```
and the you can use it in other files like this:

```
import sessionService from 'sessionService';

sessionService.createSession();
```